src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -96,6 +98,80 @@ int distance(BoundDecisionDagNode x)
             return result.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Enumerate the paths from the root node to the node of interest.
+        /// </summary>
+        /// <param name="nodes">The set of nodes in topological order.</param>
+        /// <param name="targetNode">The node of interest.</param>
+        /// <param name="nullPaths">Whether to permit following paths that test for null.</param>
+        /// <returns>Each path (excluding the node of interest) and whether it requires some when clause to evaluate to 'false'.</returns>
+        private static IEnumerable<(ImmutableArray<BoundDecisionDagNode> path, bool requiresFalseWhenClause)> GetPathsToNode(
```

Comment:
>GetPathsToNode [](http://example.com/codeflow?start=110&length=14)

It would be good to have some sort of debug or test hook to confirm that we can successfully enumerate all DAG shapes that we have tests for.

Reply:
Thanks. That caught a missing case (null)